## Assignment 1: Python group project due on Wednesday Feb. 7th @ 8AM
Use this notebook perform the following analyses using Python code and the **2018** compustat file. For the following questions, please write your answers to each question in the Canvas textbox. 

The code down below should support the answers you indicated; the code should be in an easy to follow format with comments to indicate what question you are trying to answer and what your code is accomplishing. You can also use MarkDown cells if you prefer that over commenting (you can read the tips on how to create markdown cells at the bottom of this file if you wish).

After you finish the analyses, please **make sure every cell is executed and all the returning results are presented**. Then you can save the file and submit it to Canvas by uploading it to the textbox like our first in-class assignment. **Each group only has to submit one file but everyone should make sure they are able to write and run the code.**

### Questions
- 1. Describe the dataset, *Compustat_fy2018.csv*, specifically identify:
    - How many observations are there? (1 point)
    - How many variables are there? (1 point)
    - What are the names of those variables? (1 point)

- 2. Now for Q2 and Q3, we will be performing an analysis about General Motors Co. (GM) to see whether it outperformed its industry during 2018. First, we need to find the SIC industry code for General Motors Co. (tic == 'GM'). **What is the SIC industry code for GM?** (2 points)
    
- 3. (6 points) Since performance is a construct (i.e., a concept), we pick two common measures that capture performance: Return on Equity and Asset Turnover Ratio. Calculate these two measures using the definition below and answer: did GM outperform its industry **average** (mean) in fiscal year 2018 for each of the following two measures?
    - Return on Equity, ROE: (net income /  total equity)
    - Asset turnover ratio: (revenue / total assets)
    
    *hints:*
    - *the only command we know so far that gives us an average is describe*
    - *Once you find GM's industry code, how can we limit our dataframe to just companies with the same SIC code (including GM)? Then, how can we find the average of our new variables?*
    - *You should include GM in the industry average.*

**Remember to specify your answers in the textbox below and then upload your saved and executed Jupyter Notebook to the Canvas textbox to support your answers (14 points)**

### Reference: Variable Definitions for the 2018 Compustat file
Variable Definitions in the unaltered Compustat_fy2018.csv file:
- tic (ticker symbol)
- conm (company name)
- datadate (fiscal year end)
- fyear (fiscal year)
- at (total assets)
- lt (total liabilities)
- teq (total equities)
- revt (revenue)
- ni (net income)
- exchg (numeric exchange code; 11 = New York Stock Exchange; 12 = American Stock Exchange; 14 = NASDAQ)
- sic (Industry classification code)

### Tips: 
### How to create and write in Markdown cells
- Create a new cell by clicking on the plus icon in the menu (second icon next to the "save" one).
- Select "Markdown" instead of "Code" in the drop-down menu next to the last icon (a small keyboard).
- You can type any of your explanatory texts in the markdown cells. After you finish typing, you can run the cell to have the texts presented.
- Markdown is a lightweight coding language that you can use to create rich texts with different formats and sizes. If you want to know the basic language structure, you can go to this [website](https://commonmark.org/help/).

### Command keyboard shortcuts
- There are keyboard shortcuts for major commands, such as creating a new cell, entering the edit mode, and running the cell. You can check those shortcuts by clicking on the small keyboard icon on the far right of the menu.

In [5]:
import pandas as pd
df = pd.read_csv('Compustat_fy2018.csv')
df

tic                          conm   datadate  fyear         at  \
0      AIR                      AAR CORP  31MAY2019   2018   1517.200   
1      AAL   AMERICAN AIRLINES GROUP INC  31DEC2018   2018  60580.000   
2     CECE       CECO ENVIRONMENTAL CORP  31DEC2018   2018    392.582   
3      ASA  ASA GOLD AND PRECIOUS METALS  30NOV2018   2018    196.072   
4      PNW    PINNACLE WEST CAPITAL CORP  31DEC2018   2018  17664.202   
...    ...                           ...        ...    ...        ...   
4640  GRIN             GRINDROD SHIPPING  31DEC2018   2018    447.627   
4641  CALT            CALLIDITAS THERAPE  31DEC2018   2018     72.962   
4642  NMCI    NAVIOS MARITIME CONTAINERS  31DEC2018   2018    413.527   
4643   ACA                    ARCOSA INC  31DEC2018   2018   2172.200   
4644  CTRM           CASTOR MARITIME INC  30SEP2018   2018      9.624   

             lt       teq       revt        ni  exchg     sic  
0       611.300   905.900   2051.800     7.500   11.0  5080.0  
1     60749.000  -169.000  44541.000  1412.000   14.0  4512.0  
2       214.022   178.560    337.339    -7.121   14.0  3564.0  
3         1.238   194.834      1.629   -48.789   11.0  6799.0  
4     12315.497  5348.705   3691.247   511.047   11.0  4911.0  
...         ...       ...        ...       ...    ...     ...  
4640    155.124   292.503    319.018   -20.640   14.0  4412.0  
4641      3.403    69.559      0.000   -14.859   14.0  2834.0  
4642    231.054   182.473    133.921    12.700   14.0  4412.0  
4643    487.700  1684.500   1460.400    75.700   11.0  3440.0  
4644      0.149     9.475      3.961     0.981   14.0  4412.0  

[4645 rows x 11 columns]

In [12]:
num_observations = len(df)
num_observations

4645

In [13]:
num_variables = df.shape[1]
num_variables

11

As we see from just saying df - there are 4645 observations and 11 variables 

In [15]:
variable_names = df.columns.tolist()
variable_names

['tic',
 'conm',
 'datadate',
 'fyear',
 'at',
 'lt',
 'teq',
 'revt',
 'ni',
 'exchg',
 'sic']

The variable names are

['tic',
 'conm',
 'datadate',
 'fyear',
 'at',
 'lt',
 'teq',
 'revt',
 'ni',
 'exchg',
 'sic']

In [18]:
gm_data = df[df['tic'] == 'GM']
gm_sic_code = gm_data['sic'].iloc[0]
gm_sic_code

3711.0

GMs sic code is 3711

In [22]:
gm_net_income = gm_data['ni'].iloc[0]
gm_net_income

8014.0

In [23]:
gm_total_equity = gm_data['teq'].iloc[0]
gm_total_equity

42777.0

In [24]:
gm_roe = gm_net_income / gm_total_equity
gm_roe

0.18734366598873226

GMs roe is 18.7%

In [27]:
gm_revenue = gm_data['revt'].iloc[0]
gm_revenue

147049.0

In [28]:
gm_total_assets = gm_data['at'].iloc[0]
gm_total_assets

227339.0

In [29]:
gm_asset_turnover_ratio = gm_revenue / gm_total_assets
gm_asset_turnover_ratio

0.6468269852511008

GMs asset turnover is 64.7%

In [36]:
industry_data = df[df['sic'] == gm_data['sic'].iloc[0]]

In [37]:
industry_average_roe = industry_data['ni'].sum() / industry_data['teq'].sum()
industry_average_roe

0.090673382825267

In [38]:
industry_average_asset_turnover_ratio = industry_data['revt'].sum() / industry_data['at'].sum()
industry_average_asset_turnover_ratio

0.7046699151776161

The industry average for ROE is 9.1%
The industry average for asset turnover is 70.5%

# Results

GM - roe = 18.7%

GM - asset turnover = 64.7%

Industry Average - roe = 9.1%

Industry Average - asset turnover = 70.5%